routing/result_interpretation.go

```diff
@@ -367,7 +367,19 @@ func (i *interpretedResult) processPaymentOutcomeUnknown(route *route.Route) {
 // failNode marks the node indicated by idx in the route as failed. This
 // function intentionally panics when the self node is failed.
 func (i *interpretedResult) failNode(rt *route.Route, idx int) {
+	// Mark the node as failing.
 	i.nodeFailure = &rt.Hops[idx-1].PubKeyBytes
+
+	// Mark the incoming connection as failed for the node.
+	inPair, _ := getPair(rt, idx-1)
+	i.pairResults[inPair.Reverse()] = pairResult{}
+
+	// If not the ultimate node, mark the outgoing connection as failed for
+	// the node.
```

Comment:
If we have `A <-> B <-> C` and `B` failed, the incoming is ` A -> B` and outgoing is `B -> C`?

Reply:
Yes. Added a helper variables to make this clearer.